File: ptcg-server-java/src/main/java/com/ptcg/server/model/card/basic/EnergyCard.java
package com.ptcg.server.model.card.basic;

import lombok.Data;
import lombok.EqualsAndHashCode;
import java.util.ArrayList;
import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
public abstract class EnergyCard extends Card {

    private EnergyType energyType = EnergyType.BASIC;
    private List<CardType> provides = new ArrayList<>();
    private String text = "";

    @Override
    public SuperType getSuperType() {
        return SuperType.ENERGY;
    }
}

File: ryuu-play/packages/common/src/store/card/energy-card.ts
import { Card } from './card';
import { SuperType, CardType, EnergyType } from './card-types';


export abstract class EnergyCard extends Card {

  public superType: SuperType = SuperType.ENERGY;

  public setNumber: string = '';
  
  public energyType: EnergyType = EnergyType.BASIC;

  public provides: CardType[] = [];

  public text: string = '';
}
